src/main/java/com/intuit/graphql/orchestrator/schema/transform/FieldResolverTransformerPostMerge.java

```diff
@@ -118,15 +122,65 @@ private FieldResolverContext updateWithTargetFieldData(final FieldResolverContex
     List<ResolverArgumentDefinition> updatedResolverArgumentDefinitions = updateWithInputType(
             resolverDirectiveDefinition.getArguments(), targetFieldInputValueDefinitions, fieldResolverContext);
 
+    FieldContext targetFieldContext = new FieldContext(getParentTypeName(targetFieldDefinition), targetFieldDefinition.getName());
+
+    //if it contains '.' must find correct targetContext to get namespace since it is nested
+    String namespace = (StringUtils.contains(resolverDirectiveDefinition.getField(), FQN_FIELD_SEPARATOR)) ?
+            getTargetNamespace(fieldResolverContext, resolverDirectiveDefinition, sourceXtextGraph) :
+            sourceXtextGraph.getCodeRegistry().get(targetFieldContext).getNamespace();
+
     String targetFieldFQN = resolverDirectiveDefinition.getField();
     return fieldResolverContext.transform(builder -> builder
             .targetFieldDefinition(targetFieldDefinition)
-            .targetFieldContext(new FieldContext(getParentTypeName(targetFieldDefinition), targetFieldDefinition.getName()))
+            .targetFieldContext(targetFieldContext)
+            .targetServiceNamespace(namespace)
             .resolverDirectiveDefinition(new ResolverDirectiveDefinition(targetFieldFQN, updatedResolverArgumentDefinitions))
     );
 
   }
 
+  private String getTargetNamespace(final FieldResolverContext fieldResolverContext,
+                                    final ResolverDirectiveDefinition resolverDirectiveDefinition,
+                                    final UnifiedXtextGraph sourceXtextGraph) {
+    String namespace = fieldResolverContext.getServiceNamespace();
+    String[] resolverPath = resolverDirectiveDefinition.getField().split("\\.");
+    AtomicReference<String> parentTypeNameRef = new AtomicReference<>(Operation.QUERY.getName());
+    int start = (FQN_KEYWORD_QUERY.equals(resolverPath[0]) || Operation.QUERY.getName().equals(resolverPath[0])) ? 1 : 0;
+
+    //traverse through path to get the correct fieldContext
+    for (int i = start; i < resolverPath.length; i++) {
+      String fieldName = resolverPath[i];
+      FieldContext fieldContext = new FieldContext(parentTypeNameRef.get(), fieldName);
+      DataFetcherContext dfContext = sourceXtextGraph.getCodeRegistry().get(fieldContext);
+      if(dfContext != null) {
+        namespace = dfContext.getNamespace();
+      }
+
+      ObjectTypeDefinition currentParent = sourceXtextGraph.getObjectTypeDefinitions().get(parentTypeNameRef.get());
+      if(currentParent != null) {
+        String currentFieldsTypeName =  currentParent.getFieldDefinition()
```

Comment:
getFieldDefinitionFromObjectTypeDefinition should be util. Also, we might be doing this at a lot of places.

Reply:
Gonna add a refactor case to use getFieldDefinitionFromObjectTypeDefinition for all the other locations and move the xtext util methods to the other repo